server/src/main/java/org/jboss/as/server/controller/git/ElytronClientSshdSessionFactory.java

```diff
@@ -0,0 +1,237 @@
+import org.eclipse.jgit.errors.UnsupportedCredentialItem;
+import org.eclipse.jgit.internal.transport.sshd.CachingKeyPairProvider;
+import org.eclipse.jgit.transport.CredentialItem;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.URIish;
+import org.eclipse.jgit.transport.sshd.IdentityPasswordProvider;
+import org.eclipse.jgit.transport.sshd.KeyPasswordProvider;
+import org.eclipse.jgit.transport.sshd.SshdSession;
+import org.eclipse.jgit.transport.sshd.SshdSessionFactory;
+import org.eclipse.jgit.util.FS;
+import org.jboss.as.server.logging.ServerLogger;
+import org.wildfly.client.config.ConfigXMLParseException;
+import org.wildfly.security.auth.callback.CredentialCallback;
+import org.wildfly.security.auth.client.AuthenticationConfiguration;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.AuthenticationContextConfigurationClient;
+import org.wildfly.security.auth.client.ElytronXmlParser;
+import org.wildfly.security.credential.KeyPairCredential;
+import org.wildfly.security.credential.PasswordCredential;
+import org.wildfly.security.credential.SSHCredential;
+import org.wildfly.security.password.interfaces.ClearPassword;
+
+
+/**
+ * SshdSessionFactory for JGIt using Elytron.
+ * Supports specifying ssh-keys as {@link KeyPairCredential} or {@link SSHCredential}
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+public class ElytronClientSshdSessionFactory extends SshdSessionFactory {
+
+    private static AuthenticationContextConfigurationClient CLIENT = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
+    private final AuthenticationContext context;
+
+    private final Map<Tuple, Iterable<KeyPair>> defaultKeys;
+    private String[] defaultIdentities;
+    private KeyPair keyPair;
+    private URI uri;
+    private String knownHostsFile;
+
+    public ElytronClientSshdSessionFactory(URI authenticationConfig) throws ConfigXMLParseException, GeneralSecurityException {
+        if (authenticationConfig != null) {
+            context = ElytronXmlParser.parseAuthenticationClientConfiguration(authenticationConfig).create();
+        } else {
+            context = null;
+        }
+
+        this.defaultKeys = new ConcurrentHashMap();
+        this.setSshDirectory(SSHCredential.DEFAULT_SSH_DIRECTORY);
+        this.defaultIdentities = SSHCredential.DEFAULT_PRIVATE_KEYS;
+        this.keyPair = null;
+        this.knownHostsFile = SSHCredential.DEFAULT_KNOWN_HOSTS;
+    }
+
+    @Override
+    @NonNull
+    protected List<Path> getDefaultIdentities(@NonNull File sshDir) {
+        return (List) Arrays.asList(this.defaultIdentities).stream().map((s) -> {
+            return (new File(sshDir, s)).toPath();
+        }).filter((var0) -> {
+            return Files.exists(var0, new LinkOption[0]);
+        }).collect(Collectors.toList());
+    }
```

Comment:
Overall in this case it could be written something like:
```
return Stream.of(defaultIdentities)
        .map((s) -> sshDir.toPath().resolve(s))
        .filter(path -> Files.exists(path))
        .collect(Collectors.toList());
```

Reply:
Updated, thanks!